cpu/lpc2387/periph/timer.c

```diff
@@ -136,19 +153,49 @@ int timer_init(tim_t tim, unsigned long freq, timer_cb_t cb, void *arg)
 
 int timer_set_absolute(tim_t tim, int channel, unsigned int value)
 {
-    if (((unsigned) tim >= TIMER_NUMOF) || ((unsigned) channel >= TIMER_CHAN_NUMOF)) {
+    if (((unsigned) tim >= TIMER_NUMOF) || ((unsigned) channel >= TIMER_CHANNELS)) {
         return -1;
     }
 
     lpc23xx_timer_t *dev = get_dev(tim);
+
+    set_oneshot(tim, channel);
+
     dev->MR[channel] = value;
+    /* Match Interrupt */
     dev->MCR |= (1 << (channel * 3));
     return 0;
 }
 
+int timer_set_periodic(tim_t tim, int channel, unsigned int value, unsigned flags)
+{
+    if (((unsigned) tim >= TIMER_NUMOF) || ((unsigned) channel >= TIMER_CHANNELS)) {
+        return -1;
+    }
+
+    lpc23xx_timer_t *dev = get_dev(tim);
+
+    if (flags & TIM_FLAG_RESET_ON_SET) {
+        /* reset the timer */
+        dev->TCR = 2;
+        /* start the timer */
+        dev->TCR = 1;
```

Comment:
That's not redundant, both are `volatile`.
The first one stops & resets the timer.
The second one starts the timer and clears reset.

How do I get `cppcheck` to accept that?

Reply:
Out of curiosity: Did you check if `dev->TCR = 3` would reset it and start it right away with the new value?

Reply:
Yes that did not work - the timer will remain in RESET as long as BIT2 is set.